tests/functional/products/test_config_db_share.py

```diff
@@ -0,0 +1,220 @@
+                                                        "root:root")
+        self._pr_client_2 = env.setup_product_client(
+            self.test_workspace_secondary, session_token=root_token_2)
+        self.assertIsNotNone(self._pr_client_2)
+
+    def test_read_product_from_another_server(self):
+        """
+        Test if adding and removing a product is visible from the other server.
+        """
+
+        # Check if the main server's product is visible on the second server.
+        self.assertEqual(
+            self._pr_client_2.getProducts('producttest', None)[0].endpoint,
+            'producttest',
+            "Main server's product was not loaded by the secondary server.")
+
+        # Create a new product on the secondary server.
+        name = base64.b64encode('producttest_second')
+        product_cfg = ProductConfiguration(
+            endpoint='producttest_second',
+            displayedName_b64=name,
+            description_b64=name,
+            connection=DatabaseConnection(
+                engine='sqlite',
+                host='',
+                port=0,
+                username_b64='',
+                password_b64='',
+                database=os.path.join(self.test_workspace_secondary,
+                                      'data.sqlite'))
+        )
+
+        self.assertTrue(self._pr_client_2.addProduct(product_cfg),
+                        "Cannot create product on secondary server.")
+
+        # Use the same CodeChecker config that was used on the main server,
+        # but store into the secondary one.
+        store_cfg = deepcopy(self.test_cfg['codechecker_cfg'])
+        store_cfg.update({
+            'viewer_port': self.codechecker_cfg_2['viewer_port'],
+            'viewer_product': self.codechecker_cfg_2['viewer_product']})
+        codechecker.login(store_cfg,
+                          self.test_workspace_secondary,
+                          'root', 'root')
+        store_res = codechecker.store(store_cfg,
+                                      'test_proj-secondary',
+                                      os.path.join(self.test_workspace_main,
+                                                   'reports'))
+        self.assertEqual(store_res, 0, "Storing the test project failed.")
+
+        cc_client_2 = env.setup_viewer_client(self.test_workspace_secondary)
+        self.assertEqual(len(cc_client_2.getRunData(None)), 1,
+                         "There should be a run present in the new server.")
+
+        self.assertEqual(len(self._cc_client.getRunData(None)), 1,
+                         "There should be a run present in the database when "
+                         "connected through the main server.")
+
+        # Remove the product through the main server.
+        p_id = self._root_client.getProducts('producttest_second', None)[0].id
+        p_id2 = self._pr_client_2.getProducts('producttest_second', None)[0].id
```

Comment:
How about a test if we try to run a store not on the main server where the product was removed?

Reply:
Added a test for that.